tests/Unit/classes/PrestaShopAutoloadTest.php

```diff
@@ -48,6 +48,26 @@ public function testGenerateIndex()
         $this->assertEquals($data['OrderControllerCore']['path'], 'controllers/front/OrderController.php');
     }
 
+    public function testGenerateIndexWithoutOverride()
+    {
+        \Configuration::updateGlobalValue('PS_DISABLE_OVERRIDES', 1);
+        @mkdir(_PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'override/classes/', 0777, true);
+        define('_PS_HOST_MODE_', 1);
+        file_put_contents(_PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'override/classes/Connection.php',
+            '<?php 
+            class Connection extends ConnectionCore {
+        }');
```

Comment:
Shouldn't this file be cleaned up afterwards?

Reply:
look at tearDownAfterClass() ;)